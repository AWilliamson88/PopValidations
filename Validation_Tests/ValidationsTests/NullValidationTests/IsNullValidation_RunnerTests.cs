using ApprovalTests;
using FluentAssertions;
using PopValidations;
using PopValidations_Tests.TestHelpers;
using PopValidations_Tests.ValidationsTests.GenericTestableObjects;
using System.Threading.Tasks;
using Xunit;

namespace PopValidations_Tests.ValidationsTests
{
    public class IsNull_NoError_TestingValidator : AbstractValidator<NullAllFieldTypesDto>
    {
        public IsNull_NoError_TestingValidator()
        {
            Describe(x => x.Integer).IsNull();
            Describe(x => x.String).IsNull();
            Describe(x => x.Decimal).IsNull();
            Describe(x => x.Double).IsNull();
            Describe(x => x.Short).IsNull();
            Describe(x => x.Long).IsNull();
            Describe(x => x.TwoComponentTuple).IsNull();
            Describe(x => x.TwoComponentNewTupple).IsNull();
            Describe(x => x.AllFieldTypesList).IsNull();
            Describe(x => x.AllFieldTypesLinkedList).IsNull();
            Describe(x => x.AllFieldTypesIEnumerable).IsNull();
            Describe(x => x.AllFieldTypesDictionary).IsNull();
            Describe(x => x.Struct).IsNull();
        }
    }

    public class IsNullValidation_RunnerTests
    {
        [Fact]
        public async Task GivenAValidator_WithNoErrors_ThenNoErrorsAreProduced()
        {
            // Arrange
            var runner = ValidationRunnerHelper.BasicRunnerSetup(new IsNull_NoError_TestingValidator());

            // Act
            var validationResult = await runner.Validate(new NullAllFieldTypesDto()
            {
                Integer = null,
                String = null,
                Decimal = null,
                Double = null,
                Short = null,
                Long = null,
                TwoComponentTuple = null,
                TwoComponentNewTupple = null,
                AllFieldTypesList = null,
                AllFieldTypesLinkedList = null,
                AllFieldTypesIEnumerable = null,
                AllFieldTypesDictionary = null,
                Struct = null,
            });

            // Assert
            validationResult.Results.Should().BeEmpty();
        }

        [Fact]
        public async Task GivenAValidator_WithErrors_ThenEveryFieldHasErrors()
        {
            // Arrange
            var runner = ValidationRunnerHelper.BasicRunnerSetup(new IsNull_NoError_TestingValidator());

            // Act
            var validationResult = await runner.Validate(new NullAllFieldTypesDto());
            var json = JsonConverter.ToJson(validationResult);

            // Assert
            validationResult.Results.Should().HaveCount(13);
            Approvals.VerifyJson(json);
        }

        [Fact]
        public void GivenAValidator_WhenDescribing_ThenEveryFieldHasDescriptions()
        {
            // Arrange
            var runner = ValidationRunnerHelper.BasicRunnerSetup(new EqualTo_NoError_TestingValidator());

            // Act
            var description = runner.Describe();
            var json = JsonConverter.ToJson(description);

            // Assert
            description.Results.Should().HaveCount(7);
            Approvals.VerifyJson(json);
        }
    }
}
