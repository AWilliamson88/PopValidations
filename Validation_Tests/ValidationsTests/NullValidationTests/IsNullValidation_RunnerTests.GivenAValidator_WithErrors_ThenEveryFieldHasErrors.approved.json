{
    "results": [
        {
            "property": "Integer",
            "outcomes": [
                {
                    "isFailed": true,
                    "isSucceeded": false,
                    "propertyName": "Integer",
                    "message": "Is null",
                    "group": null
                }
            ]
        },
        {
            "property": "String",
            "outcomes": [
                {
                    "isFailed": true,
                    "isSucceeded": false,
                    "propertyName": "String",
                    "message": "Is null",
                    "group": null
                }
            ]
        },
        {
            "property": "Decimal",
            "outcomes": [
                {
                    "isFailed": true,
                    "isSucceeded": false,
                    "propertyName": "Decimal",
                    "message": "Is null",
                    "group": null
                }
            ]
        },
        {
            "property": "Double",
            "outcomes": [
                {
                    "isFailed": true,
                    "isSucceeded": false,
                    "propertyName": "Double",
                    "message": "Is null",
                    "group": null
                }
            ]
        },
        {
            "property": "Short",
            "outcomes": [
                {
                    "isFailed": true,
                    "isSucceeded": false,
                    "propertyName": "Short",
                    "message": "Is null",
                    "group": null
                }
            ]
        },
        {
            "property": "Long",
            "outcomes": [
                {
                    "isFailed": true,
                    "isSucceeded": false,
                    "propertyName": "Long",
                    "message": "Is null",
                    "group": null
                }
            ]
        },
        {
            "property": "TwoComponentTuple",
            "outcomes": [
                {
                    "isFailed": true,
                    "isSucceeded": false,
                    "propertyName": "TwoComponentTuple",
                    "message": "Is null",
                    "group": null
                }
            ]
        },
        {
            "property": "TwoComponentNewTupple",
            "outcomes": [
                {
                    "isFailed": true,
                    "isSucceeded": false,
                    "propertyName": "TwoComponentNewTupple",
                    "message": "Is null",
                    "group": null
                }
            ]
        },
        {
            "property": "AllFieldTypesList",
            "outcomes": [
                {
                    "isFailed": true,
                    "isSucceeded": false,
                    "propertyName": "AllFieldTypesList",
                    "message": "Is null",
                    "group": null
                }
            ]
        },
        {
            "property": "AllFieldTypesLinkedList",
            "outcomes": [
                {
                    "isFailed": true,
                    "isSucceeded": false,
                    "propertyName": "AllFieldTypesLinkedList",
                    "message": "Is null",
                    "group": null
                }
            ]
        },
        {
            "property": "AllFieldTypesIEnumerable",
            "outcomes": [
                {
                    "isFailed": true,
                    "isSucceeded": false,
                    "propertyName": "AllFieldTypesIEnumerable",
                    "message": "Is null",
                    "group": null
                }
            ]
        },
        {
            "property": "AllFieldTypesDictionary",
            "outcomes": [
                {
                    "isFailed": true,
                    "isSucceeded": false,
                    "propertyName": "AllFieldTypesDictionary",
                    "message": "Is null",
                    "group": null
                }
            ]
        },
        {
            "property": "Struct",
            "outcomes": [
                {
                    "isFailed": true,
                    "isSucceeded": false,
                    "propertyName": "Struct",
                    "message": "Is null",
                    "group": null
                }
            ]
        }
    ]
}